test/format/format_config.c

```diff
@@ -1449,6 +1451,88 @@ config_statistics(void)
     }
 }
 
+/*
+ * config_chunk_cache --
+ *     Chunk cache configuration.
+ */
+static void
+config_chunk_cache(void)
+{
+    char buf[128], chunkcache_storage_path[512];
+    const char *chunkcache_type;
+
+    chunkcache_type = NULL;
+
+    /* Chunkcache does not work unless tiered storage is configured. */
+    if (!g.tiered_storage_config) {
+        if (config_explicit(NULL, "chunk_cache") && GV(CHUNK_CACHE))
+            testutil_die(EINVAL,
+              "%s: chunkcache cannot be enabled unless tiered storage is configured.", progname);
+        return;
+    }
+
+    if (!config_explicit(NULL, "chunk_cache")) {
+        /*
+         * Make sure no configurations related to chunk caching are set if chunkcache is not
+         * enabled.
+         */
+        if (config_explicit(NULL, "chunk_cache.capacity") ||
+          config_explicit(NULL, "chunk_cache.chunk_size") ||
+          config_explicit(NULL, "chunk_cache.type") ||
+          config_explicit(NULL, "chunk_cache.storage_path"))
+            testutil_die(EINVAL,
+              "%s: Enable chunk caching (chunk_cache=on) to allow configuring other chunk cache "
+              "settings",
+              progname);
+
+        /* Enable chunkcache 50% of the time if not explicit set. */
+        testutil_snprintf(
+          buf, sizeof(buf), "chunk_cache=%s", mmrand(&g.data_rnd, 1, 100) <= 50 ? "on" : "off");
+        config_single(NULL, buf, false);
+    }
+
+    if (GV(CHUNK_CACHE)) {
+        if (config_explicit(NULL, "chunk_cache.type")) {
+            chunkcache_type = GVS(CHUNK_CACHE_TYPE);
+            if (strcmp(chunkcache_type, "FILE") != 0 && strcmp(chunkcache_type, "DRAM") != 0)
+                testutil_die(EINVAL, "illegal chunkcache.type configuration: %s", chunkcache_type);
+
+            if (GV(RUNS_IN_MEMORY) && strcmp(chunkcache_type, "FILE") == 0)
+                testutil_die(EINVAL,
+                  "%s: chunk caching cannot be enabled for in-memory runs as chunkcache.type is "
+                  "set to FILE.",
+                  progname);
+        } else {
+            if (GV(RUNS_IN_MEMORY))
```

Comment:
Do we know how often we enable in-memory runs, and is it useful to test DRAM when we aren't doing in memory runs?

Reply:
We enable in-memory runs 5% of the time in format. Therefore, I think that for the remaining 95% of the time, it is reasonable to have a 50% distribution for each.